### DISCUSSION TITLE
V4: Cannot search documents when Resource source is "Documents"

### DISCUSSION THREAD

**Discussion by KeiBontakun** [microsoft-search/pnp-modern-search#1514] [Q&A] [answered]
I have installed PnP Modern Search V4(4.4.1) and got successful search when "Resource source ID" is "Local SharePointResults".
However when "Documents" is selected, search does not work with any keywords and always every files are shown.
Is there some settings required for "Documents"?
Please help, thanks.

Figure 1: Resource source is "LocalSharePointResults" and 2 files are shown successfully for query "pdf".
![LocalSharePointResults](https://user-images.githubusercontent.com/3262436/141447524-d732f7df-6a5a-4dc6-84e1-535cd86494aa.jpg)

Figure 2: Resource source is "Documents" and every (52) files are shown for "pdf".
![Documents](https://user-images.githubusercontent.com/3262436/141447509-d3770b16-3383-4669-a1cc-135e8fb0b5c1.jpg)

**Comment by wobba** [comment, marked as answer]
I would advice from not using the classic predefined result sources and rather add the required KQL manually. You can always check the definition of the classic sources and see if they return semantically what you expect.

**Reply by KeiBontakun** [reply]
Thanks, I modified the **Query template** like following and got wanted results.
`{searchTerms} path:"https://example.sharepoint.com/sites/example/Shared Documents/" FileType:(pdf OR doc* OR xls* OR ppt*)`

**Reply by wobba** [reply]
Love your use of `FileType:(...)` btw 🙂
